playground/src/kata.tsx

```diff
@@ -42,15 +84,34 @@ export function Kata(props: {
 
     props.kata.sections.forEach((section, idx) => {
       const parentDiv = itemContent.current[idx];
-      const div = parentDiv?.querySelector(".kata-item-content");
-      if (!div) return;
-      if (section.type === "text") {
-        div.innerHTML = section.contentAsHtml;
-      } else if (section.type === "exercise") {
-        div.innerHTML = section.solutionAsHtml;
-      } else {
-        div.innerHTML = "";
+      let titlePrefix = "\u{1F41B}";
+      if (section.type === "exercise") {
+        titlePrefix = "\u{2328} Exercise: ";
+        const descriptionDiv = parentDiv?.querySelector(
+          ".exercise-description"
+        );
+        if (!descriptionDiv) return;
+        descriptionDiv.innerHTML = section.description.asHtml;
+        const solutionDiv = parentDiv?.querySelector(".exercise-solution");
+        if (!solutionDiv) return;
+        solutionDiv.innerHTML = ExplainedSolutionAsHtml(
+          section.explainedSolution
+        );
+      } else if (section.type === "lesson") {
+        titlePrefix = "\u{1F4D6} Lesson: ";
+        const contentDiv = parentDiv?.querySelector(".kata-item-content");
+        if (!contentDiv) return;
```

Comment:
All these early returns should probably be `throw` statements if the expectation is they should never fail.

Reply:
Using throw statements now.